src/ArborX_HDBSCAN.hpp

```diff
@@ -12,31 +12,44 @@
 #define ARBORX_HDBSCAN_HPP
 
 #include <ArborX_Dendrogram.hpp>
+#include <ArborX_DetailsKokkosExtScopedProfileRegion.hpp>
 #include <ArborX_MinimumSpanningTree.hpp>
 
 namespace ArborX::Experimental
 {
 
 template <typename ExecutionSpace, typename Primitives>
 auto hdbscan(ExecutionSpace const &exec_space, Primitives const &primitives,
-             int core_min_size)
+             int core_min_size,
+             DendrogramImplementation dendrogram_impl =
+                 DendrogramImplementation::DEFAULT)
 {
-  Kokkos::Profiling::pushRegion("ArborX::HDBSCAN");
+  KokkosExt::ScopedProfileRegion guard("ArborX::HDBSCAN");
 
   using MemorySpace = typename Primitives::memory_space;
 
-  Kokkos::Profiling::pushRegion("ArborX::HDBSCAN::mst");
-  Details::MinimumSpanningTree<MemorySpace> mst(exec_space, primitives,
-                                                core_min_size);
-  Kokkos::Profiling::popRegion();
+  if (dendrogram_impl == DendrogramImplementation::DEFAULT)
+    dendrogram_impl = DendrogramImplementation::BORUVKA;
 
-  Kokkos::Profiling::pushRegion("ArborX::HDBSCAN::dendrogram");
-  Dendrogram<MemorySpace> dendrogram(exec_space, mst.edges);
-  Kokkos::Profiling::popRegion();
+  if (dendrogram_impl != DendrogramImplementation::BORUVKA)
+  {
+    Kokkos::Profiling::pushRegion("ArborX::HDBSCAN::mst");
+    Details::MinimumSpanningTree<MemorySpace> mst(exec_space, primitives,
+                                                  core_min_size);
+    Kokkos::Profiling::popRegion();
 
-  Kokkos::Profiling::popRegion();
+    Kokkos::Profiling::pushRegion("ArborX::HDBSCAN::dendrogram");
+    Dendrogram<MemorySpace> dendrogram(exec_space, mst.edges);
+    Kokkos::Profiling::popRegion();
 
-  return dendrogram;
+    return dendrogram;
+  }
+
+  // Hybrid Boruvka+dendrogram
+  Details::MinimumSpanningTree<MemorySpace, true> mst(exec_space, primitives,
```

Comment:
What does `true` mean? Can this be named?

Reply:
This is part of the interface discussion I want to have. I'm not the biggest fan of having a boolean template parameter to MST, and welcome any suggestions.

Reply:
The difference is that in addition to the constructing MST (`edges` member variable), we also construct the dendrogram (`dendrogram_parents` and `dendrogram_parent_heights` (which are really just edge weights)). To construct, we also do some extra computations that we don't need for just MST.

I would be find creating some `MSTBase` class, and then having MST alias to non-HDBSCAN mode, and use the base inside HDBSCAN*.

Reply:
Since `MinimumSpanningTree` is just in `Details` and therefore only supposed to be used within the library, I would be fine to just give the option a name via an enum class.
If we decide to move it out of `Details`, we could reconsider if the option needs to be user-facing.